### PR TITLE
no parking signs on bus lanes.

### PR DESCRIPTION
**Pull request by kianzarrin**
fixed: parking must not be configurable on bus lanes. 

[TMPE.zip](https://ci.appveyor.com/api/projects/krzychu124/tmpe/artifacts/TMPE.zip?branch=bus-parking-signs)

Before:
image
After:
image

### CODE REVIEW

File: TLM/TLM/Manager/Impl/ParkingRestrictionsManager.cs [hunk 1]
```diff
@@ -25,15 +25,29 @@ public class ParkingRestrictionsManager
...
             }
 
             ItemClass connectionClass = segment.Info.GetConnectionClass();
-
             return connectionClass.m_service == ItemClass.Service.Road && segment.Info.m_hasParkingSpaces;
         }
 
+        public bool MayHaveParkingRestriction(ushort segmentId, NetInfo.Direction finalDir) {
+            ref NetSegment segment = ref segmentId.ToSegment();
+            bool right = (finalDir == NetInfo.Direction.Forward) != segment.m_flags.IsFlagSet(NetSegment.Flags.Invert);
```
Comment (krzychu124) [inline TLM/TLM/Manager/Impl/ParkingRestrictionsManager.cs:38]: What about LHT _left-hand traffic_(RHD - _right hand drive_)?

Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/ParkingRestrictionsManager.cs:38]: I use final dir for that reason
I need to test that though. I only tested invert

Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/ParkingRestrictionsManager.cs:38]: thanks for reminding me. it does not work 
image

File: TLM/TLM/Manager/Impl/ParkingRestrictionsManager.cs [hunk 2]
```diff
@@ -25,15 +25,30 @@ public class ParkingRestrictionsManager
...
 
             ItemClass connectionClass = segment.Info.GetConnectionClass();
-
             return connectionClass.m_service == ItemClass.Service.Road && segment.Info.m_hasParkingSpaces;
         }
 
+        public bool MayHaveParkingRestriction(ushort segmentId, NetInfo.Direction finalDir) {
+            ref NetSegment segment = ref segmentId.ToSegment();
+            var dir = Shortcuts.RHT ? finalDir : NetInfo.InvertDirection(finalDir);
+            bool right = dir == NetInfo.Direction.Forward != segment.m_flags.IsFlagSet(NetSegment.Flags.Invert);
```
Comment (krzychu124) [inline TLM/TLM/Manager/Impl/ParkingRestrictionsManager.cs:39]: That's a little bit confusing. I don't like double equality check with mixed types

Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/ParkingRestrictionsManager.cs:39]: should I put parenthesis?
`bool != bool`  is xor
`bool == bool` is xnor

**Review by kvakvs — APPROVED** [review]

**Pull request merged**